gcp_variant_transforms/transforms/merge_headers.py

```diff
@@ -13,26 +13,174 @@
+      # Unknown number of values
+      return None
+    else:
+      raise ValueError('Incompatible numbers cannot be resolved: '
+                       '{}, {}'.format(first, second))
+
+  def _is_bigquery_field_repeated(self, vcf_num):
+    """Returns true if the corresponding field in bigquery schema is repeated.
+
+    Args:
+      vcf_num (integer): value of field `Number` in VCF header.
+    """
+    if vcf_num in (0, 1):
+      return False
+    elif vcf_num == -1 and self._split_alternate_allele_info_fields:
+      # info field with `Number=A` (represented by -1) does not become
+      # a repeated field if flag `split_alternate_allele_info_fields` is on.
+      # See `variant_transform_options.py` for more details.
+      return False
+    else:
+      return True
+
 
+class HeaderMerger(object):
+  """Class for merging two vcf headers."""
+
+  def __init__(self, resolver):
+    self._resolver = resolver
+
+  def merge(self, first, second):
+    """Updates ``first``'s headers with values from ``second``.
+
+    If a specific key does not already exist in a specific one of ``first``'s
+    headers, that key and the associated value will be added. If the key does
+    already exist in the specific header of ``first``, then the value of that
+    key will be updated with the value from ``second``.
+
+    Args:
+      first (:class:`VcfHeader`): The VcfHeader object.
+      second (:class:`VcfHeader`): The VcfHeader object that's headers will be
+        merged into the headers of first.
+    """
+    if not isinstance(first, VcfHeader) or  not isinstance(first, VcfHeader):
+      raise NotImplementedError
+
+    self._merge_header_fields(first.infos, second.infos)
+    self._merge_header_fields(first.filters, second.filters)
+    self._merge_header_fields(first.alts, second.alts)
+    self._merge_header_fields(first.formats, second.formats)
+    self._merge_header_fields(first.contigs, second.contigs)
+
+  def _merge_header_fields(self, first, second):
+    """Modifies ``first`` to add any keys from ``second`` not in ``first``.
+
+    Args:
+      first (dict): first header fields.
+      second (dict): second header fields.
+
+    Raises:
+      ValueError: If the header fields are incompatible (e.g. same key with
+        different types or numbers).
+    """
+    for second_key, second_value in second.iteritems():
+      if second_key not in first:
+        first[second_key] = second_value
+        continue
+      first_value = first[second_key]
+      if first_value.keys() != second_value.keys():
+        raise ValueError('Incompatible header fields: {}, {}'.format(
+            first_value, second_value))
+      merged_value = OrderedDict()
+      for first_field_key, first_field_value in first_value.iteritems():
+        second_field_value = second_value[first_field_key]
+        if first_field_value == second_field_value:
+          merged_value.update({first_field_key : first_field_value})
```

Comment:
nit: remove space before ":" (not enforced by pylint, but it's enforced by pep8...hopefully we can integrate that to our repo too).

Reply:
Done